crates/oxc_transformer/src/typescript/class.rs

```diff
@@ -22,29 +33,90 @@ impl<'a> TypeScript<'a, '_> {
     ///   this.y = y;
     /// }
     /// ```
-    pub(super) fn transform_class(class: &mut Class<'a>, ctx: &mut TraverseCtx<'a>) {
+    ///
+    /// 2. Convert class fields to `this` assignments in the constructor body.
+    ///
+    /// > This transformation only works when set_public_class_fields is true and the fields have initializers,
+    ///   which is to align with the behavior of TypeScript's useDefineForClassFields: false option.
+    ///
+    /// Input:
+    /// ```ts
+    /// class C {
+    ///   x = 1;
+    ///   [y] = 2;
+    /// }
+    /// ```
+    ///
+    /// Output:
+    /// ```js
+    /// let _y;
+    /// class C {
+    ///   static {
+    ///     _y = y;
+    ///   }
+    ///   constructor() {
+    ///     this.x = 1;
+    ///     this[_y] = 2;
+    ///   }
+    /// }
+    /// ```
+    ///
+    // The computed key transformation behavior is the same as `TypeScript`, computed key assignments are
+    // inserted into a static block rather than Babel that inserts them before class. We follow `TypeScript` just for
+    // simplicity because Babel handles class expression and class declaration differently, which quite troublesome to
+    // implement. Anyway, `TypeScript` is the source of truth for the typescript transformation.
+    //
+    // TODO: Not handling static property yet. Might no need to handle it because static property wouldn't affect the
+    //       constructor body, which means never breaks https://github.com/oxc-project/oxc/issues/9192. And the
+    //       `class-properties` plugin has covered it.
+    pub(super) fn transform_class(&self, class: &mut Class<'a>, ctx: &mut TraverseCtx<'a>) {
```

Comment:
Could you take a look at this? Should we support transforming static property key for `useDefineForClassFields: false`?

Reply:
I guess static properties also needs to be transformed.
For this input,
```ts
class Base {
    static set foo(v: unknown) { console.log('set!') }
}
class Foo extends Base {
    static foo = 0;
}
```
TypeScript outputs
```js
"use strict";
class Base {
    static set foo(v) { console.log('set!'); }
}
class Foo extends Base {
    static { this.foo = 0; }
}
```
and this outputs `set!`.

If this is not transformed, `set!` won't be output.

Reply:
Another case that is also good to test is:
```ts
const fields = {
  get name() {
    console.log("field")
    return "field"
  }
}

class Foo {
    static {
      console.log('before')
    }

    static [fields.name] = 0;

    static {
      console.log('after')
    }
}
```
this should output `field`, `before`, `after`.

Reply:
Thank you for reviewing and bringing this case up! My brain is small, I hadn't thought of this case.😢. Yes, you are right, we should also transform static properties.